tests/queries/0_stateless/02985_variant_type_analytical.sh

```diff
@@ -0,0 +1,108 @@
+{
+    create_table "$1"
+
+    $CH_CLIENT -q  "insert into test values(1, 4);"
+    $CH_CLIENT -q  "insert into test values(2, 2);"
+    $CH_CLIENT -q  "insert into test values(3, 1);"
+    $CH_CLIENT -q  "insert into test values(4, 5);"
+    $CH_CLIENT -q  "insert into test values(5, 3);"
+
+    $CH_CLIENT -q  "insert into test values(6, 'b');"
+    $CH_CLIENT -q  "insert into test values(7, 'd');"
+    $CH_CLIENT -q  "insert into test values(8, 'a');"
+    $CH_CLIENT -q  "insert into test values(9, 'c');"
+    $CH_CLIENT -q  "insert into test values(10, 'e');"
+
+    $CH_CLIENT -q  "insert into test values(11, 12.0);"
+    $CH_CLIENT -q  "insert into test values(12, 13.5);"
+    $CH_CLIENT -q  "insert into test values(13, 15.0);"
+    $CH_CLIENT -q  "insert into test values(14, 13.0);"
+    $CH_CLIENT -q  "insert into test values(15, 15.20);"
+    $CH_CLIENT -q  "insert into test values(16, 11.0);"
+    $CH_CLIENT -q  "insert into test values(17, 15.25);"
+    $CH_CLIENT -q  "insert into test values(18, 14.0);"
+
+    #$CH_CLIENT -nmq "select id, v, variantType(v) from test order by v;" # Does not sort
+
+    $CH_CLIENT -nmq "select id, v, variantType(v) from test order by v.UInt64;"
+
+    $CH_CLIENT -nmq "select id, v, variantType(v) from test order by v.String;"
+
+    $CH_CLIENT -nmq "select id, v, variantType(v) from test order by v.Float64;"
+
+    $CH_CLIENT -nmq "select id, v, variantType(v) from test order by v.UInt64, v.String, v.Float64;"
+
+    $CH_CLIENT -nmq "select id, v, variantType(v) from test order by v.UInt64 desc, v.String desc, v.Float64 desc;"
+}
+
+function test_group_by()
+{
+    create_table "$1"
+
+    $CH_CLIENT -q  "insert into test values(1, 1);"
+    $CH_CLIENT -q  "insert into test values(2, 1);"
+    $CH_CLIENT -q  "insert into test values(3, 1);"
+    $CH_CLIENT -q  "insert into test values(4, 2);"
+    $CH_CLIENT -q  "insert into test values(5, 2);"
+
+    $CH_CLIENT -q  "insert into test values(6, 'a');"
+    $CH_CLIENT -q  "insert into test values(7, 'a');"
+    $CH_CLIENT -q  "insert into test values(8, 'b');"
+    $CH_CLIENT -q  "insert into test values(9, 'b');"
+    $CH_CLIENT -q  "insert into test values(10, 'b');"
+
+    $CH_CLIENT -q  "insert into test values(11, '2');"
+
+    $CH_CLIENT -q  "insert into test values(12, 12.0);"
+    $CH_CLIENT -q  "insert into test values(13, 12.0);"
+    $CH_CLIENT -q  "insert into test values(14, 13.25);"
+    $CH_CLIENT -q  "insert into test values(15, 13.25);"
+    $CH_CLIENT -q  "insert into test values(16, 13.25);"
+    $CH_CLIENT -q  "insert into test values(17, 14.5);"
+    $CH_CLIENT -q  "insert into test values(18, 14.5);"
+    $CH_CLIENT -q  "insert into test values(19, 15.025);"
+    $CH_CLIENT -q  "insert into test values(20, 15.025);"
+
+    $CH_CLIENT -nmq "select v, variantType(v), count(*) from test.test group by v;"
+
+    #$CH_CLIENT -nmq "select id, v, variantType(v), max(id) over (partition by v) as max_id from test;" # bucketing is wrong
```

Comment:
Probably it's because `Variant` is not comparable, I will investigate. I think it should work, good catch!